ASTProcessor/src/test/java/edu/usb/argos/ASTProcessor/visitor/classes/JavaClassStructureCollectorTest.java

```diff
@@ -0,0 +1,56 @@
+package edu.usb.argos.ASTProcessor.visitor.classes;
+
+import edu.usb.argos.ASTProcessor.antlr.JavaLexer;
+import edu.usb.argos.ASTProcessor.visitor.core.services.collectors.classes.JavaClassStructureCollector;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import edu.usb.argos.ASTProcessor.antlr.JavaParser;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class JavaClassStructureCollectorTest {
+    private JavaClassStructureCollector structureCollector;
+    private JavaParser.CompilationUnitContext compilationUnit;
+
+    @BeforeEach
+    void setUp() {
+        String testClass =
+                "public class TestClass extends BaseClass implements Interface1, Interface2 {\n" +
+                        "    private String field;\n" +
+                        "    public void method() {}\n" +
+                        "}";
+
+        CharStream input = CharStreams.fromString(testClass);
+        JavaLexer lexer = new JavaLexer(input);
+        CommonTokenStream tokens = new CommonTokenStream(lexer);
+        JavaParser parser = new JavaParser(tokens);
+        compilationUnit = parser.compilationUnit();
+
+        structureCollector = new JavaClassStructureCollector();
+    }
+
+    @Test
+    void getSuperClass_ShouldReturnCorrectSuperClass() {
```

Comment:
Use CamelCase for test name, don't snake test and the same with the rest

Reply:
Done